dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/GitHubSourceCodeRepo.java

```diff
@@ -231,8 +233,12 @@ public List<String> listFiles(String repositoryId, String pathToDirectory, Strin
         }
     }
 
-    private String readFileFromRepo(String fileName, String reference, GHRepository repo) {
+    private String readFileFromRepo(final String originalFileName, final String originalReference, final GHRepository originalRepo) {
```

Comment:
Some JavaDoc could be helpful; I had a hard time understanding this method, especially with the loop that resets the index on line 261/288.

Is it doing this?

* Read file from `repo/folder1/folder2/foo.wdl`
* Convert path from `repo/submodule/foo.wdl` to read from `submodulerepo/foo.wdl`
* Convert path from `repo/symlink/goo.wdl` to read from `repo/actualpath/goo.wdl`

Reply:
Yeah, it is also looking directory by directory to keep things in the cache. 
For example `repo/folder1` may be in the cache and `repo/folder1/folder2` may be in the cache. If you go to `repo/folder1/folder2/foo.wdl` directly, you could 404 and use up a request. But looking at the contents of the directories may let you know if it exists without using up a request

Reply:
Added